tsfile/src/main/java/org/apache/iotdb/tsfile/write/writer/tsmiterator/TSMIterator.java

```diff
@@ -0,0 +1,154 @@
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iotdb.tsfile.write.writer.tsmiterator;
+
+import org.apache.iotdb.tsfile.file.metadata.ChunkGroupMetadata;
+import org.apache.iotdb.tsfile.file.metadata.ChunkMetadata;
+import org.apache.iotdb.tsfile.file.metadata.IChunkMetadata;
+import org.apache.iotdb.tsfile.file.metadata.TimeseriesMetadata;
+import org.apache.iotdb.tsfile.file.metadata.enums.TSDataType;
+import org.apache.iotdb.tsfile.file.metadata.statistics.Statistics;
+import org.apache.iotdb.tsfile.read.common.Path;
+import org.apache.iotdb.tsfile.utils.Pair;
+import org.apache.iotdb.tsfile.utils.PublicBAOS;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import java.util.TreeMap;
+
+/**
+ * TSMIterator returns full path of series and its TimeseriesMetadata iteratively. It accepts data
+ * source from memory or disk. Static method getTSMIteratorInMemory returns a TSMIterator that reads
+ * from memory, and static method getTSMIteratorInDisk returns a TSMIterator that reads from disk.
+ */
+public class TSMIterator implements Iterator<Pair<String, TimeseriesMetadata>> {
+  private static Logger LOG = LoggerFactory.getLogger(TSMIterator.class);
+  protected List<Pair<Path, List<IChunkMetadata>>> sortedChunkMetadataList;
+  protected Iterator<Pair<Path, List<IChunkMetadata>>> iterator;
+
+  protected TSMIterator(List<ChunkGroupMetadata> chunkGroupMetadataList) {
+    this.sortedChunkMetadataList = sortChunkMetadata(chunkGroupMetadataList, null, null);
+    this.iterator = sortedChunkMetadataList.iterator();
+  }
+
+  public static TSMIterator getTSMIteratorInMemory(
+      List<ChunkGroupMetadata> chunkGroupMetadataList) {
+    return new TSMIterator(chunkGroupMetadataList);
+  }
+
+  public static TSMIterator getTSMIteratorInDisk(
+      File cmtFile, List<ChunkGroupMetadata> chunkGroupMetadataList, LinkedList<Long> serializePos)
+      throws IOException {
+    return new DiskTSMIterator(cmtFile, chunkGroupMetadataList, serializePos);
+  }
+
+  @Override
+  public boolean hasNext() {
+    return iterator.hasNext();
+  }
+
+  @Override
+  public Pair<String, TimeseriesMetadata> next() {
+    Pair<Path, List<IChunkMetadata>> nextPair = iterator.next();
+    try {
+      return new Pair<>(
+          nextPair.left.getFullPath(),
+          constructOneTimeseriesMetadata(nextPair.left.getMeasurement(), nextPair.right));
+    } catch (IOException e) {
+      LOG.error("Meets IOException when getting next TimeseriesMetadata", e);
+      return null;
```

Comment:
should not ignore the IOException, we should rethrow it and let the upper layer to know that.